kivy/logger.py

```diff
@@ -150,47 +151,36 @@ class FileHandler(logging.Handler):
     encoding = 'utf-8'
 
     def purge_logs(self):
-        '''Purge log is called randomly to prevent the log directory from being
-        filled by lots and lots of log files.
-        You've a chance of 1 in 20 that purge log will be fired.
-        '''
-        if randint(0, 20) != 0:
-            return
-        if not self.log_dir:
-            return
```

Comment:
Please add back this check. It's needed when Kivy is running in certain configurations where there's no config.

Reply:
This must be checked *before* you import config.

Reply:
Check for self.log_dir is now issued before the import of config. Unittests were also created. (Including test logs in the test_logger subfolder)